### PR TITLE
Created Quantum Walks Tutorial

### PR DESCRIPTION
**Pull request by Lucaman99**
Sorry for opening another PR, but the CLA in the other request simply won't go through. This takes into account a lot of the changes requested in #1934.

### CODE REVIEW

**Comment by Lucaman99** [comment]
Also @Strilanc, I made all of the requested changes to built-in Cirq functions, however, wasn't able to find documentation for the Cirq ArithmeticOperations class. As a result, this is the one change I haven't made yet, is this a fairly new feature?

**Comment by vtomole** [comment]
@Lucaman99 Sign the CLA

**Comment by Lucaman99** [comment]
I have, but it's saying that "co-contributors" to this commit haven't signed it, which is likely due to me accidentally using a different email when I made some of the commits a couple months back

**Comment by Lucaman99** [comment]
"We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for all the commit author(s) or Co-authors."

**Comment by Lucaman99** [comment]
@googlebot I fixed it.

**Comment by Lucaman99** [comment]
Squash all the commits previous to the most recent one?

**Comment by vtomole** [comment]
@Lucaman99 Yes, then change the commit email to the primary one on your GitHub account (https://stackoverflow.com/a/3042512/5716192).

When you `Cirq$ git config --list`, is the user.email the primary one in your Github account?

**Comment by Lucaman99** [comment]
Ok, will do!

File: examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,766 @@
...
+    "# Quantum Random Walks With Cirq - Tutorial"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Classical Random Walks**\n",
+    "\n",
+    "A random walks is a random process involving a \"walker\" that is placed in some $n$-dimensional medium \n",
```
Comment (vtomole) [inline examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb]: Change `walks` -> `walk`.

File: examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb [hunk 2]
```diff
@@ -0,0 +1,766 @@
...
+    "(like a grid or a graph). We then repeatedly query some random variable, and based on the outcome of our measurement,\n",
```
Comment (vtomole) [inline examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb]: Get rid of parentheses.

File: examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb [hunk 3]
```diff
@@ -0,0 +1,766 @@
...
+    "the walker's position vector (position on the graph or grid) is updated. A very basic example of a random walk is \n",
```
Comment (vtomole) [inline examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb]: Get rid of `very`

File: examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb [hunk 4]
```diff
@@ -0,0 +1,766 @@
...
+    "the one-dimensional graphical case, where we consider a marker placed on the origin of a number line with markings\n",
+    "at each of the integers. Let the initial position vector of our marker be $\\lvert 0\\rangle$. For $N$ steps of our\n",
+    "random walk, so take a set of $N$ random variables, $\\{X_1, \\ ..., \\ X_N\\}$, which can take on either a value of \n",
```
Comment (vtomole) [inline examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb]: The `so take` after the comma is awkward to read. Maybe get rid of `so`

File: examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb [hunk 5]
```diff
@@ -0,0 +1,766 @@
...
+    "$1$ or $-1$ with equal probability ($0.5$). To find the updated position vector of our walker, we simply compute\n",
```
Comment (vtomole) [inline examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb]: Get rid of `(0.5)`.

File: examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb [hunk 6]
```diff
@@ -0,0 +1,766 @@
...
+    "\n",
+    "$$j \\ = \\ \\displaystyle\\sum_{k \\ = \\ 1}^{N} \\ X_k$$\n",
+    "\n",
+    "Where we know:\n",
+    "\n",
+    "\n",
+    "$$\\lvert \\text{Final}\\rangle \\ = \\ \\lvert \\text{Initial} \\ + \\ j\\rangle$$\n",
+    "\n",
+    "\n",
+    "So for our case, the final position vector is simply $\\lvert j\\rangle$. This model of a random walk can easily \n",
```
Comment (vtomole) [inline examples/.ipynb_checkpoints/random_walk_tutorial-checkpoint.ipynb]: Get rid of `simply` and `easily`

File: examples/notebook_tutorials/random_walk_tutorial.ipynb [hunk 7]
```diff
@@ -0,0 +1,943 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Quantum Walks With Cirq - Tutorial"
+   ]
+  },
```
Comment (vtomole) [inline examples/notebook_tutorials/random_walk_tutorial.ipynb]: Add a blurb about what is about what this notebook about and also `To get started, we'll need to import these libraries`. Something like that.

File: examples/notebook_tutorials/random_walk_tutorial.ipynb [hunk 8]
```diff
@@ -0,0 +1,943 @@
...
+    "$$\\lvert \\text{Final}\\rangle \\ = \\ \\lvert \\text{Initial} \\ + \\ j\\rangle$$\n",
+    "\n",
+    "\n",
+    "So for our case, the final position vector is simply $\\lvert j\\rangle$. This model of a random walk can easily \n",
+    "be generalized to $n$-dimensions. \n",
+    "\n",
+    "\n",
+    "Another important fact to note is that for a discrete, 1-dimensional random \n",
+    "walk on a number-line-like graph, the probability of the random walker being at a specific location follows \n",
+    "a binomial distribution. Let us define an $N$-step random walk. Let us then assert that $N \\ = \\ L \\ + \\ R$, \n",
```
Comment (vtomole) [inline examples/notebook_tutorials/random_walk_tutorial.ipynb]: Link to https://en.wikipedia.org/wiki/Binomial_distribution

File: examples/notebook_tutorials/random_walk_tutorial.ipynb [hunk 9]
```diff
@@ -0,0 +1,943 @@
...
+    "where $L$ is the number of steps to the left, and $R$ is the number of steps to the right. We can then reason \n",
+    "that if there is some probability $p_{r}$ of the walker taking a rightward step at one time-step of the random \n",
+    "walk, the probability of taking a leftward step is given by $1 \\ - \\ p_{r}$. It follows that the probability \n",
+    "of taking $L \\ = \\ N \\ - \\ R$ leftward steps, and $R$ rightward steps  in a random walk of $N$ total steps is \n",
+    "given by:\n",
```
Comment (vtomole) [inline examples/notebook_tutorials/random_walk_tutorial.ipynb]: Get rid of `given by`.

**Comment by vtomole** [comment]
@Lucaman99 I'm still working my way through this. So far:

-  Get rid of all the `simple`s and `easy`s. I haven't seen that it's necessary in any of your sentences.
- Add a `references` section at the end  for people who want to go deeper into random walks.

**Comment by Lucaman99** [comment]
Great, I'll make these changes! Thank you!

**Comment by Lucaman99** [comment]
Hey @vtomole - I just made all of the requested changes, let me know what you think! 🙂

**Comment by Lucaman99** [comment]
Thanks for your response @vtomole - I'll make those changes tonight 🙁

File: examples/notebook_tutorials/random_walk_tutorial.ipynb [hunk 10]
```diff
@@ -0,0 +1,960 @@
...
+    "Where the upward-arrow symbol represent counter-clockwise motion, and the downward arrow represents \n",
+    "clock-wise motion. Now that we have defined all of the vectors we need to encode the information about \n",
+    "our random walk, we must understand how we can realize these vectors in our quantum algorithm. Well, this is \n",
+    "again fairly simple. For a graph of $K \\ = \\ 2^n$ nodes, we require $n$ qubits to encode binary representations \n",
+    "of numbers ranging from $0$ to $K \\ - \\ 1$, therefore each of the vectors spanning $H_W$ will be given \n",
+    "by the binary representation of $j$ corresponding to the basis vector $\\lvert j\\rangle$. For the coin vector, \n",
+    "since we have only two states, we only need one qubit to encode the two possible states:\n",
+    "\n",
+    "\n",
+    "$$\\lvert 0\\rangle \\ = \\ \\lvert \\uparrow\\rangle \\ \\ \\text{and} \\ \\
```
Comment (vtomole) [inline examples/notebook_tutorials/random_walk_tutorial.ipynb]: Where do you actually use this function in this tutorial? If you don't it should be removed

Comment (vtomole) [inline examples/notebook_tutorials/random_walk_tutorial.ipynb]: Remove reference to your self. We don't know who 'i' is. You never told us before saying this.

Comment (vtomole) [inline examples/notebook_tutorials/random_walk_tutorial.ipynb]: > whcih

Did you run a spell and grammar checker on all of this? Please do. I'm not going to be able to catch all the little things like this.

File: examples/notebook_tutorials/random_walk_tutorial.ipynb [hunk 11]
```diff
@@ -0,0 +1,960 @@
...
+    "    # Repeatedly queries our random variable and moves our walker, for the specified number of steps\n",
+    "    \n",
+    "    for j in range(0, N): \n",
+    "    \n",
+    "        coin_flip = list(np.random.choice(2, 1, p=[1-pr, pr])) #Flips our weighted coin\n",
+    "        position += 2*coin_flip[0]-1 #Moves our walker according to the coin flip \n",
+    "    \n",
+    "    return position\n",
+    "    \n",
+    "print(\"The walker is located at: x = \"+str(random_walk(pr, N, i)))"
```
Comment (vtomole) [inline examples/notebook_tutorials/random_walk_tutorial.ipynb]: Use `.format` instead.

**Comment by Lucaman99** [comment]
Hi @vtomole! Just wanted to ping you in case you hadn't seen this yet, I made the requested changes last week. Let me know when you get the chance to take a look! 🙂

File: examples/notebook_tutorials/new_test.py [hunk 12]
```diff
@@ -0,0 +1,57 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (vtomole) [inline examples/notebook_tutorials/new_test.py]: You missed removing this file: . See https://github.com/quantumlib/Cirq/pull/2657#issuecomment-579439362

**Comment by Lucaman99** [comment]
@vtomole Sorry about that, I just deleted it

**Comment by Lucaman99** [comment]
Done!

**Comment by vtomole** [comment]
Hey @dstrain115, You'll need to manually add the cla for us to merge this.

**Comment by dstrain115** [comment]
Added the CLA manually.  Hopefully that fixes things.

You may want to make sure that all your environments are set up with the same git author so that this doesn't complain in the future.

File: examples/notebooks/nqubittoff.ipynb [hunk 13]
```diff
@@ -0,0 +1,123 @@
+{
```
Comment (vtomole) [inline examples/notebooks/nqubittoff.ipynb]: Remove this file

**Comment by Lucaman99** [comment]
Ok, done

**Comment by vtomole** [comment]
@Lucaman99 Looks like my merge voided the cla. I think this is what we should do: copy these files into a new branch and commit them with the email adress with which you signed the cla and open a new pull request.

**Comment by Lucaman99** [comment]
Sounds good, I'll do that right now

**Pull request closed without merging**